userland_generic.ld

```diff
@@ -89,6 +95,21 @@ SECTIONS {
         . = ALIGN(4); /* Make sure we're word-aligned here */
     } > FLASH =0xFF
 
+    /* Need to reserve room for the stack in the linker file. This makes the
+     * _got addresses used by the compiler match what they will be when the
+     * app is loaded into memory. This is not necessary for full PIC supported
+     * platforms (like Cortex-M), but is needed when an app is compiled for a
+     * fixed address.
```

Comment:
I don't quite understand this comment. Why are there GOT address on non-PIC platforms?

Reply:
I don't know? But there are.

Reply:
Can you provide a dump of the symbols that are in the GOT section for a RISC-V application? I'm worried that the existence of the GOT section is a sign that we're doing something wrong.

Reply:
OK I moved the `-fPIC` flag to its own category, and don't include it with the RISC-V builds. The GOT size is now 0.